prg_test.go

```diff
@@ -0,0 +1,32 @@
+package test
+
+import (
+	"math"
+	mrand "math/rand"
+	"testing"
+
+	. "github.com/bjartek/overflow"
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNextUInt64NewPRGOverflow(t *testing.T) {
+	o, err := OverflowTesting()
+	assert.NoError(t, err)
+
+	t.Run("Should generate uniform distribution", func(t *testing.T) {
+		// make sure n is a power of 2 so that there is no bias in the last class
+		// n is a random power of 2 (from 2 to 2^10)
+		n := 1 << (1 + mrand.Intn(10))
+		classWidth := (math.MaxUint64 / uint64(n)) + 1
+
+		// using the same seed, the salt varies in getNextUInt64()
+		seed := GetRandomSeed(t)
```

Comment:
The uniformity checked in this test is different from the uniformity that we expect to test when assessing a PRG quality : this test looks at the the uniformity of the first generated output across multiple PRG instances (diversified per salt), the quality of a PRG is evaluated using successive outputs of the same PRG instance (same seed and salt). 
A PRG that succeeds the first criteria may fail the second (the main quality one). 

I suggest two things:
 - update the current test to use the same PRG instance and evaluate the uniformity of it successive outputs. 
 - add another test, to make sure both the seed and salt are taken into account in the implementation. This doesn't need to be a statistical test. It's enough to check that two instances with different seeds but same salt output different randoms (the first output is enough to check), and then the same for 2 instances with the same seed but different salts.

Reply:
This is really helpful, thank you! I'll make these changes shortly.